src/AspNetCoreModuleV2/CommonLib/debugutil.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#include "debugutil.h"
+#include "dbgutil.h"
+
+inline HANDLE g_hStandardOutput;
+
+VOID
+DebugInitialize()
+{
+    g_hStandardOutput = GetStdHandle(STD_OUTPUT_HANDLE);
+
+    HKEY hKey;
+
+    if (RegOpenKeyEx(HKEY_LOCAL_MACHINE,
+            L"SOFTWARE\\Microsoft\\IIS Extensions\\IIS AspNetCore Module\\Parameters",
+            0,
+            KEY_READ,
+            &hKey) == NO_ERROR)
+    {
+        DWORD dwType;
+        DWORD dwData;
+        DWORD cbData;
+
+        cbData = sizeof(dwData);
+        if ((RegQueryValueEx(hKey,
+            L"DebugFlags",
+            NULL,
+            &dwType,
+            (LPBYTE)&dwData,
+            &cbData) == NO_ERROR) &&
+            (dwType == REG_DWORD))
+        {
+            DEBUG_FLAGS_VAR = dwData;
+        }
+
+        RegCloseKey(hKey);
+    }
+
+    // We expect single digit value and a null char
+    const size_t environmentVariableValueSize = 2;
+    std::wstring environmentVariableValue(environmentVariableValueSize, '\0');
+
+    if (GetEnvironmentVariable(L"ASPNETCORE_MODULE_DEBUG", environmentVariableValue.data(), environmentVariableValueSize) == environmentVariableValueSize - 1)
+    {
+        try
+        {
+            const auto value = std::stoi(environmentVariableValue);
+
+            if (value >= 1) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_ERROR;
+            if (value >= 2) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_WARNING;
+            if (value >= 3) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_INFO;
+            if (value >= 4) DEBUG_FLAGS_VAR |= ASPNETCORE_DEBUG_FLAG_CONSOLE;
+        }
+        catch (...)
+        {
+            // ignore
+        }
+    }
+}
+
+BOOL
+IsEnabled(
+    DWORD   dwFlag
+    )
+{
+    return ( dwFlag & DEBUG_FLAGS_VAR );
+}
+
+VOID
+DebugPrint(
+    DWORD   dwFlag,
+    const LPCSTR  szString
+    )
+{
```

Comment:
Add WDebugPrint.

Reply:
No point, if you are doing no formatting why are you logging wide string?

Reply:
tru.